pandas/core/generic.py

```diff
@@ -2342,6 +2344,11 @@ def to_json(
 
             .. versionadded:: 0.23.0
 
+        indent : integer, default 0
+           Length of whitespace used to indent each record.
+
+           .. versionadded:: 1.0.0
```

Comment:
does this match the semantics of the stdlib, e.g. default is ``None`` and 0 means insert newlines, also can be a string, is that supported? https://docs.python.org/3/library/json.html

Reply:
No this only accepts int for now and 0 does not insert new lines. This matches ujson behavior instead of stdlib

Reply:
It should be possible to support this in the long term I just think will take a little bit of effort to bridge the gap with our vendored ujson. Thinking for now it might make the most sense to change the signature to `indent=None` and have that map to `indent=0`, documenting the the `indent=0` behavior is different from stdlib. 

I think that should avoid a deprecation cycle in the future for `indent=0` behavior if we eventually can mirror the stdlib's support